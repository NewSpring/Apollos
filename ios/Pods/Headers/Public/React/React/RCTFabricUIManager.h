../../../../../../node_modules/react-native/React/Fabric/RCTFabricUIManager.h