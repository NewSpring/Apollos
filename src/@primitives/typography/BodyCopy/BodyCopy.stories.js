import React from 'react';

import { storiesOf } from '@storybook/react-native';

import BodyCopy from './';

storiesOf('@primitives/typography', module)
  .add('BodyCopy', () => (
    <BodyCopy>{'"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'}</BodyCopy>
  ));
