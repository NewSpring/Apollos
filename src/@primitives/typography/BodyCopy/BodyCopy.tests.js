import React from 'react';
import renderer from 'react-test-renderer';
import ThemeProvider from '@primitives/ThemeProvider';
import P from './';

describe('the P component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <P>Default P text</P>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <ThemeProvider>
        <P style={salmon}>Salmon text</P>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <P accessible={false}>{'"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'}</P>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
