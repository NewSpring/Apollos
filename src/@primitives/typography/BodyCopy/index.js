import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@primitives/styled';
import rem from '@utils/remUnit';
import verticalRhythm from '@utils/verticalRhythm';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
  }),
);

const StyledBodyCopy = styled(({ theme }) => ({
  fontSize: rem(1),
  lineHeight: verticalRhythm(1, 1),
  fontFamily: theme.fontFamilySerif,
  color: theme.baseFontColor,
}))(Text);

const BodyCopy = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledBodyCopy
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledBodyCopy>
));

export default BodyCopy;
