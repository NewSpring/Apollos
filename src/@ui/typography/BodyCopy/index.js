import React from 'react';
import { Text } from 'react-native';
import { compose, pure, setPropTypes } from 'recompose';
import PropTypes from 'prop-types';
import styled from '@ui/styled';

const enhance = compose(
  pure,
  setPropTypes({
    children: PropTypes.node,
    style: Text.propTypes.style,
  }),
);

const StyledBodyCopy = styled(({ theme }) => ({
  fontSize: theme.helpers.rem(1),
  lineHeight: theme.helpers.verticalRhythm(1, 1),
  fontFamily: theme.typography.fontFamilySerif,
  color: theme.colors.text.primary,
}), 'BodyCopy')(Text);

const BodyCopy = enhance(({
  children,
  style: styleProp = {},
  ...otherProps
}) => (
  <StyledBodyCopy
    style={styleProp}
    {...otherProps}
  >
    {children}
  </StyledBodyCopy>
));

export default BodyCopy;
