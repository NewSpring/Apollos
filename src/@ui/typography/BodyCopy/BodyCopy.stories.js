import React from 'react';

import { storiesOf } from '@storybook/react-native';

import BodyCopy from './';

storiesOf('@ui/typography', module)
  .add('BodyCopy', () => (
    <BodyCopy>{'"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'}</BodyCopy>
  ));
