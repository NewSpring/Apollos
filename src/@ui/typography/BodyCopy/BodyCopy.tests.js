import React from 'react';
import renderer from 'react-test-renderer';
import Providers from '@ui/TestProviders';
import BodyCopy from './';

describe('the BodyCopy component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <Providers>
        <BodyCopy>Default BodyCopy text</BodyCopy>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as bold', () => {
    const tree = renderer.create(
      <Providers>
        <BodyCopy bold>Bold BodyCopy text</BodyCopy>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as italic', () => {
    const tree = renderer.create(
      <Providers>
        <BodyCopy italic>Italic BodyCopy text</BodyCopy>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render as bold italic', () => {
    const tree = renderer.create(
      <Providers>
        <BodyCopy bold italic>Bold italic BodyCopy text</BodyCopy>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept and render passed in styles', () => {
    const salmon = { color: 'salmon' };
    const tree = renderer.create(
      <Providers>
        <BodyCopy style={salmon}>Salmon text</BodyCopy>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should accept additional props', () => {
    const tree = renderer.create(
      <Providers>
        <BodyCopy accessible={false}>{'"True faith means holding nothing back. It means putting every hope in God\'s fidelity to His Promises." â€• Francis Chan'}</BodyCopy>
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
