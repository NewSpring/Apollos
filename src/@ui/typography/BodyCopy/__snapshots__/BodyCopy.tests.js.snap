// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the P component should accept additional props 1`] = `
<Text
  accessible={false}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#505050",
      "fontFamily": "Georgia",
      "fontSize": 18,
      "lineHeight": 18,
    }
  }
>
  "True faith means holding nothing back. It means putting every hope in God's fidelity to His Promises." â€• Francis Chan
</Text>
`;

exports[`the P component should accept and render passed in styles 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "salmon",
      "fontFamily": "Georgia",
      "fontSize": 18,
      "lineHeight": 18,
    }
  }
>
  Salmon text
</Text>
`;

exports[`the P component should render correctly 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "#505050",
      "fontFamily": "Georgia",
      "fontSize": 18,
      "lineHeight": 18,
    }
  }
>
  Default P text
</Text>
`;
